pkg/clockcache/cache.go

```diff
@@ -95,11 +102,15 @@ func (self *Cache) insert(key interface{}, value interface{}) (canonicalKey inte
 		self.elementsLock.Lock()
 		defer self.elementsLock.Unlock()
 		delete(self.elements, insertLocation.key)
-		*insertLocation = element{key: key, value: value}
+		self.dataSize -= insertLocation.size
+		self.dataSize += size
+		self.evictions++
+		*insertLocation = element{key: key, value: value, size: size}
 	} else {
 		self.elementsLock.Lock()
 		defer self.elementsLock.Unlock()
-		self.storage = append(self.storage, element{key: key, value: value})
+		self.storage = append(self.storage, element{key: key, value: value, size: size})
+		self.dataSize += size
 		insertLocation = &self.storage[len(self.storage)-1]
```

Comment:
[unrelated to PR] Another question, rather a doubt. Why don't we simply store the index of the element in the storage, rather than storing its address? In that way, it's more efficient?

Reply:
I think the lookup is more efficient for pointer dereference than an array index but I'm not positive. we can ask Josh.

Reply:
If both are nearly same, then if index way is possible, then we should do that as it will make gc work a bit less.

Reply:
ToDo in this PR

Reply:
Todo for next PR.